pkg/api/pod/util_test.go

```diff
@@ -3472,3 +3472,112 @@ func TestDropClusterTrustBundleProjectedVolumes(t *testing.T) {
 		})
 	}
 }
+
+func TestDropPodLifecycleSleepAction(t *testing.T) {
+	container := api.Container{
+		Name: "foo",
+	}
+
+	containerWithHandler := func() api.Container {
+		return api.Container{
+			Name: "foo",
+			Lifecycle: &api.Lifecycle{
+				PreStop: &api.LifecycleHandler{
+					Sleep: &api.SleepAction{Seconds: 1},
+				},
+			},
+		}
+	}
+
+	podWithHandler := func() *api.Pod {
+		return &api.Pod{
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					containerWithHandler(),
+				},
+			},
+		}
+	}
+
+	podWithoutHandler := func() *api.Pod {
+		return &api.Pod{
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					container,
+				},
+			},
+		}
+	}
+
+	podInfo := []struct {
+		description    string
+		hasSleepAction bool
+		pod            func() *api.Pod
+	}{
+		{
+			description:    "has sleep action",
+			hasSleepAction: true,
+			pod:            podWithHandler,
+		},
+		{
+			description:    "does not have sleep action",
+			hasSleepAction: false,
+			pod:            podWithoutHandler,
+		},
+		{
+			description:    "is nil",
+			hasSleepAction: false,
+			pod:            func() *api.Pod { return nil },
+		},
+	}
+
+	for _, enabled := range []bool{true, false} {
```

Comment:
This loop is pretty hard to verify by reading.  I think it would be mush simpler to just say something like:


```
makeLifecycle := func(postStart, preStop *api.LifecycleHandler) *api.Lifecycle {
	return &api.Lifecycle{
		postStart: postStart,
		preStop: preStop,
	}
}
makeSleepHandler := func() *api.Lifecycle {
	return &api.Lifecycle{
		Sleep: &api.SleepAction{Seconds: 1},
	}
}
makeExecHandler := func() *api.Lifecycle {
	return &api.Lifecycle{
		Exec: &api.ExecAction{Command: "foobar"},
	}
}

cases := []struct {
	gateEnabled bool
	old *api.Lifecycle
	new *api.Lifecycle
	expectPostStart bool
	expectPreStop bool
}{{
	gateEnabled: false
	old: nil,
	new: nil,
	expectPostStart: false,
	expectPreStop: false,
}, {
	gateEnabled: false
	old: nil,
	new: makeLifecycle(nil, nil),
	expectPostStart: false,
	expectPreStop: false,
}, {
	gateEnabled: false
	old: nil,
	new: makeLifecycle(makeExecHandler(), makeExecHandler()),
	expectPostStart: true,
	expectPreStop: true,
}, {
	gateEnabled: false
	old: nil,
	new: makeLifecycle(makeSleepHandler(), makeSleepHandler()),
	expectPostStart: false,
	expectPreStop: false,
// and so forth, testing all the interesting permutations with and without the gate
}}
```

It would be more LOC but much more obvious to look at and say "yep, that's all of them".